tests/integration/server/location/location.controller.test.ts

```diff
@@ -401,4 +402,77 @@ describe('Location API', function () {
       });
     });
   });
+  describe('GET /rooms/admin', function () {
+    let response: Response;
+    context('As an unauthenticated user', function () {
+      beforeEach(function () {
+        authStub.resolves(null);
+      });
+      it('should return a 400 status', async function () {
+        authStub.rejects(new ForbiddenException());
+
+        response = await request(api)
+          .get('/api/rooms/admin');
+
+        strictEqual(response.ok, false);
+        strictEqual(response.status, HttpStatus.FORBIDDEN);
+      });
+    });
+    context('As an authenticated user', function () {
+      let result: RoomAdminResponse[];
+      beforeEach(async function () {
+        authStub.resolves(adminUser);
+        response = await request(api)
+          .get('/api/rooms/admin');
+        result = response.body;
+      });
+      it('should return a 200 status', function () {
+        strictEqual(response.status, HttpStatus.OK);
+      });
+      it('should return a nonempty array of data', function () {
+        strictEqual(Array.isArray(result), true);
+        notStrictEqual(result.length, 0);
+      });
+      it('returns all rooms in the database', async function () {
```

Comment:
For this test I ordered the rooms so that I could directly compare the result (which is ordered) to the expected rooms. Should I instead compare the number of rooms to check that I am returning all rooms in the database?

Reply:
Yeah, giving this some thought - I was wondering about mapping over the rooms and just grabbing the `id` and comparing two arrays of UUIDs. You could run both arrays of UUIDs through `Array.prototype.sort()` making the database ordering irrelevant (for this test, anyway). 

Or to put it another way - if we decide down the road that the rooms should be sorted in descending order - this test (as it currently stands) would fail....even though the data is all still there(just not in the order the test expected).

Reply:
Thanks for your advice Robert. I updated the test to check for that the expected and actual IDs are the same (commit [be12270](https://github.com/seas-computing/course-planner/pull/570/commits/be122705db69aae9614aa64066305616b0b362d6)).